packages/grafana-ui/src/components/uPlot/Plot.tsx

```diff
@@ -95,38 +107,15 @@ function initializePlot(data: AlignedDataWithGapTest, config: Options, el: HTMLD
   return new uPlot(config, data, el);
 }
 
-function updateData(frame: DataFrame, config: UPlotConfigBuilder, plotInstance?: uPlot, data?: AlignedData | null) {
+function updateData(
+  frame: DataFrame,
+  config: UPlotConfigBuilder,
+  plotInstance?: uPlot,
+  data?: AlignedDataWithGapTest | null
+) {
   if (!plotInstance || !data) {
     return;
   }
   pluginLog('uPlot core', false, 'updating plot data(throttled log!)', data);
-  updateScales(frame, config, plotInstance);
   plotInstance.setData(data);
 }
-
-function updateScales(frame: DataFrame, config: UPlotConfigBuilder, plotInstance: uPlot) {
```

Comment:
this is not needed since the scale ranges are all defined as functions

Reply:
Great!